### PR TITLE
fixing hidden/protected examples

### LINKED ISSUE #339: modify examples for model hidden/protected properties
## URL of the page 

http://loopback.io/doc/en/lb3/Model-definition-JSON-file.html#hidden-properties

## Nature of the issue 

examples show the hidden/protected properties nested inside the properties object, but according to the code datasource-juggler/model.js hidden/protected should be defined as top level properties

### PR DESCRIPTION
**Pull request by ebarault**
fixes #339 
see further inlined comments

### CODE REVIEW

File: pages/en/lb2/Model-definition-JSON-file.md [hunk 1]
```diff
@@ -580,20 +583,23 @@ Author will have personal information such as social security number etc, and th
...
     "email": {
       "type": "string",
       "required": true
     },
-...
-   "protected": ["email"],
-...
+    ...
+  },
+  "protected": ["email"],
```
Comment (ebarault) [inline pages/en/lb2/Model-definition-JSON-file.md:600]: well actually user.json has not protected property : should we fix this in loopback or change the example? 
cc: @bajtos

Comment (bajtos) [reply pages/en/lb2/Model-definition-JSON-file.md:600]: >  well actually user.json has not protected property : should we fix this in loopback or change the example?

I don't fully understand the purpose of `protected` properties. @raymondfeng what's your opinion?

If I had to decide, I'd rather fix the example. E.g.

```diff
- An example of a protected property is User.email:
+ An example configuring `email` as a protected property:
```

**Review by bajtos — APPROVED** [review]
Nice, thank you for the pull request @ebarault!

**Comment by ebarault** [comment]
the difference between hidden and protected properties is quite subtle and confuses quite a lot of people i'm afraid
- **hidden**: whether the property will be returned when the query directly targets the model
- **protected** : whether the property will be returned when the model is targeted is an included relation

password could be protected for sure, but alone. I'll go for your proposal @bajtos in the meantime

**Comment by crandmck** [comment]
I'll go ahead and merge this, but I think it's worth clarifying the diff. btw hidden & protected properties since it's a source of confusion.  I'll do that separately.

**Comment by ebarault** [comment]
@crandmck : thanks
please make sure to modify this
```diff
- An example of a protected property is User.email:
+ An example configuring `email` as a protected property:
```

**Comment by crandmck** [comment]
ah ok, thx for the heads up..  I'll do that

**Pull request merged**